ledger/boxtxn_test.go

```diff
@@ -369,3 +393,157 @@ func TestBoxRW(t *testing.T) {
 		dl.txn(withBr)
 	})
 }
+
+func TestBoxIOBudgets(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	genBalances, addrs, _ := ledgertesting.NewTestGenesis()
+	ledgertesting.TestConsensusRange(t, boxVersion, 0, func(t *testing.T, ver int, cv protocol.ConsensusVersion) {
+		dl := NewDoubleLedger(t, genBalances, cv)
+		defer dl.Close()
+
+		appIndex := dl.fundedApp(addrs[0], 1_000_000, boxAppSource)
```

Comment:
@jannotti I opened a child PR with optional test modifications in https://github.com/jannotti/go-algorand/pull/16.  There's _no_ correctness changes.

Can you have a look to see which (if any) tests you'd like to add here?